src/main/java/edu/byu/ece/rapidSmith/design/subsite/CellDesign.java

```diff
@@ -504,37 +617,78 @@ public void unplaceCell(Cell cell) {
 		if (cell.getDesign() != this)
 			throw new Exceptions.DesignAssemblyException("Cannot unplace cell not in the design.");
 
-		unplaceCell_impl(cell);
+		// for macros, unplace all internal cells
+		if (cell.isMacro()) {
+			cell.getInternalCells().forEach(this::unplaceCell_impl);
+		}
+		else {
+			unplaceCell_impl(cell);
+		}
 	}
 
 	private void unplaceCell_impl(Cell cell) {
-		assert cell.getDesign() == this;
-		assert cell.isPlaced();
-
+		
```

Comment:
Can we document that cell should be a leaf cell in this call.  Either a comment or an assertion